src/trans/gpu/internal/fsc_mod.F90

```diff
@@ -132,7 +132,7 @@ SUBROUTINE FSC(ALLOCATOR,HFSC,PREEL_COMPLEX, KF_FS, KF_UV, KF_SCALARS, KUV_OFFSE
     DO JM=0,R_NSMAX !(note that R_NSMAX <= G_NMEN(IGLG) for all IGLG)
       IGLG    = OFFSET_VAR+KGL-1
       IF (JM <= G_NMEN(IGLG)) THEN
-        IOFF_LAT = KF_FS*D_NSTAGTF(KGL)
+        IOFF_LAT = 1_JPIB*KF_FS*D_NSTAGTF(KGL)
```

Comment:
In the end, I think I am fine with everything, and I can change the casts if needed, as long as it is as consistent as possible. Your suggestions won't work, though. It would need to be
```
        IOFF_LAT = INT(KF_FS,JPIB)*INT(D_NSTAGTF(KGL), JPIB)
```
Or, if there is a left-to-right guarantee, then
```
        IOFF_LAT = INT(KF_FS,JPIB)*D_NSTAGTF(KGL)
```
which then again is very close to my syntax. My preference for 1_JPIB*a*b*c is that if you have several lines, the actual multiplications are not cluttered with integer conversions. 

Which one would you prefer?

Reply:
I think my concern is academic in any case. Whatever happens on the right-hand side of the assignment operator, it is casted to `JPIB` when assigning to `IOFF_LAT` anyway. So I think we can just leave this as it is.

Reply:
I think my assumption is that multiplications are evaluated left-to-right, if that is the case, then we are good, and everything is always multiplied as JPIB. If this is not the case, then we have to cast each individual operand to JPIB before we execute the multiplication, which is possible, but doesn't help readability. I hope this argument makes sense :D

Reply:
Let's just merge but keep this in mind!